### ISSUE TITLE
Language detection calls English resumes Italian, is_english() is always False, and translation chunks exceed the API limit

### ISSUE THREAD

**Issue by MOHITKOURAV01** [Muskankr/AI-Resume-Analyzer#914]
`analyzer/language_detector.py` labels ordinary English resumes as Italian, and `is_english()` returns `False` for every text it is given. `analyzer/translation_service.py` then hands the caller chunks larger than the limit the chunking exists to respect.

## English is detected as Italian

```python
>>> from analyzer.language_detector import LanguageDetector
>>> LanguageDetector.detect("Software engineer. Python, Django, React. Built APIs. Led a team of four.")
LanguageDetectionResult(language_code='it', language_name='Italian', confidence=0.42, method_used='heuristic')
```

Three things combine:

1. `langdetect` is not installed and is not in `requirements.txt`, so the `except ImportError` branch fires on every call and the heuristic is the only path that ever runs in production. The `method_used: "langdetect"` value documented in `multilingual_serializers.py` is unreachable.
2. `LANGUAGE_INDICATORS` has word lists for `es`, `fr`, `de` and `it`. There is no list for `en`. English can therefore never *win* the vote — it is only the default when every other language scores zero.
3. The Italian list contains `a`, `e`, `i`, `in`, `no`, `si`, `ha`, `ma`, `per`, `come`. Those are English words too. Any English paragraph of reasonable length scores several points for Italian and zero for English, so Italian wins outright.

## `is_english()` is always False

```python
>>> LanguageDetector.is_english("This is clearly an English sentence.")
False
```

`is_english` requires `language_code == "en" and confidence > 0.6`. Every path that returns `"en"` returns a confidence below that ceiling: the short-text fallback returns `0.0`, the empty-word fallback `0.5`, and the `max_score == 0` default `0.5`. The one branch that can produce a confidence above `0.6` is the branch that has just concluded the text is *not* English. The function cannot return `True`.

The knock-on is that `LanguageDetectionResponseSerializer.is_english` is `False` on every response, so `LanguageDetectionBanner` offers to translate resumes that are already in English.

## Ties are broken by dict insertion order

`max(scores, key=scores.get)` returns the first key at the maximum. `LANGUAGE_INDICATORS` is ordered `es, fr, de, it`, so a text scoring 2 for Spanish and 2 for French is always called Spanish, with no signal that the call was a coin flip.

## Chunks come back over the limit

```python
>>> TranslationService(use_mock=True)._chunk_text("A" * 4500 + "\n\n" + "B" * 4500)
>>> [len(c) for c in _]
[4500, 4500]
```

`MAX_CHUNK_SIZE` is 4000. `_chunk_text` splits on `\n\n` and packs paragraphs into chunks, but a single paragraph longer than the limit is never split — it is assigned whole to `current_chunk` and emitted at that size. A resume with one long unbroken block, which is what `pdfplumber` produces for a dense one-column layout, goes to the translation API in one oversized request and is rejected there.

Covered by `analyzer/tests/test_multilingual.py`, which asserts all of this and has never run — see #913.

## What "fixed" looks like

- An English word list, so English is detected rather than defaulted to.
- Scoring that measures each language against the same text and reports a confidence that means something, with a deterministic tie-break.
- `is_english()` true for English text.
- `langdetect` used when it is installed and seeded so two runs agree; the heuristic staying correct when it is not.
- `_chunk_text` splitting an oversized paragraph on sentence, then line, then hard character boundaries, and never returning a chunk above `MAX_CHUNK_SIZE`.

**Comment by Atharv094** [comment]
Hello! I’d love to tackle the `is_english()` logic and the `_chunk_text` overflow bug. For `analyzer/language_detector.py`, I’d add the missing English heuristics to fix the misclassification and ensure `is_english()` returns true. Could you please assign this to me? Is this part of the ECSoC program? If so, could an admin add the "EcSoc26" label?
